File: fingerprints/src/main/java/com/simprints/fingerprints/activities/matching/MatchingContract.java
package com.simprints.fingerprints.activities.matching;


import android.content.Intent;

import com.simprints.id.activities.BasePresenter;
import com.simprints.id.activities.BaseView;


interface MatchingContract {

    interface View extends BaseView<Presenter> {

        void setIdentificationProgress(int progress);

        void setVerificationProgress();

        void setIdentificationProgressLoadingStart();

        void setIdentificationProgressMatchingStart(int matchSize);

        void setIdentificationProgressReturningStart();

        void setIdentificationProgressFinished(int returnSize, int tier1Or2Matches, int tier3Matches, int tier4Matches, int matchingEndWaitTimeMillis);

        void launchAlert();

        void makeToastMatchNotRunning(String text);

        void doSetResult(int resultCode, Intent resultData);

        void doFinish();
    }

    interface Presenter extends BasePresenter {

    }
}

File: fingerprint/src/main/java/com/simprints/fingerprints/activities/matching/MatchingContract.kt
package com.simprints.fingerprints.activities.matching


import android.content.Intent

import com.simprints.id.activities.BasePresenter
import com.simprints.id.activities.BaseView


interface MatchingContract {

    interface View : BaseView<Presenter> {

        fun setIdentificationProgress(progress: Int)

        fun setVerificationProgress()

        fun setIdentificationProgressLoadingStart()

        fun setIdentificationProgressMatchingStart(matchSize: Int)

        fun setIdentificationProgressReturningStart()

        fun setIdentificationProgressFinished(returnSize: Int, tier1Or2Matches: Int, tier3Matches: Int, tier4Matches: Int, matchingEndWaitTimeMillis: Int)

        fun launchAlert()

        fun makeToastMatchFailed()

        fun doSetResult(resultCode: Int, resultData: Intent)

        fun doFinish()
    }

    interface Presenter : BasePresenter {

        fun dispose()
    }
}
